data_transfer/devices/dreem.py

```diff
@@ -88,53 +96,56 @@ def download_metadata(self) -> None:
                 )
             )
 
-            # Reformat to mirror UCAM/DMP.
-            patient_id = patient_id.replace("-", "") if patient_id else patient_id
-
-            if patient_id and (ucam_entry := ucam.get_record(patient_id)):
-                known += 1
-                dreem_devices = [
-                    d for d in ucam_entry.devices if self.device_type in d.device_id
-                ]
-
-                # Best-case: only one device was worn and UCAM knows it
-                if len(dreem_devices) == 1:
-                    device_record = dreem_devices[0]
-                # Edge-case: multiple dreem headbands used, e.g., if one broke.
-                elif len(dreem_devices) > 1:
-                    # Determine usage based on weartime as it exists in UCAM
-                    device_record = ucam.record_by_wear_period_in_list(
-                        dreem_devices, recording.start, recording.end
-                    )
-                # Edge-case: device not logged with patient in UCAM
-                else:
-                    print(f"Metadata cannot be determined for:\n    {recording}")
-                    continue
-            else:
+            if not (patient_id := utils.format_id_patient(_patient_id)):
```

Comment:
I think this is good to understand where errors come from. Although isolated, this will allow us to detect possible formatting errors; even though our validator will soon allow a bit more lossy written ids.